zhaquirks/tuya/ts0601_trv.py

```diff
@@ -931,19 +922,210 @@ async def command(
 ZONNSMART_ONLINE_MODE_ENUM_ATTR = 0x0473  # device publises value as enum datatype
 ZONNSMART_ONLINE_MODE_BOOL_ATTR = 0x0173  # but expects to receive bool datatype
 
+ZONNSMART_SCHEDULE_TYPE_ATTR = 0x041F  # See ZONNSMARTScheduleType
+ZONNSMART_SCHEDULE_SET_ATTR = 0x006A
+ZONNSMART_SCHEDULE_MONDAY_ATTR = 0x006C
+ZONNSMART_SCHEDULE_TUESDAY_ATTR = 0x0070
+ZONNSMART_SCHEDULE_WEDNESDAY_ATTR = 0x006D
+ZONNSMART_SCHEDULE_THURSDAY_ATTR = 0x071
+ZONNSMART_SCHEDULE_FRIDAY_ATTR = 0x006E
+ZONNSMART_SCHEDULE_SATURDAY_ATTR = 0x0072
+ZONNSMART_SCHEDULE_SUNDAY_ATTR = 0x006F
+
 ZONNSMART_MAX_TEMPERATURE_VAL = 3000
 ZONNSMART_MIN_TEMPERATURE_VAL = 500
-ZonnsmartManuClusterSelf = None
+
+
+class ZONNSMARTHolidayStartStop(t.CharacterString):
+    """Stop and start time of holidays."""
+
+    def __new__(cls, value):
+        """Create a string representation of the attribute."""
+        if isinstance(value, str):
+            return super().__new__(cls, value)
+        else:
+            s = "".join(chr(x) for x in value)[::-1]
+            humanstr = (
+                s[0:4]
+                + "/"
+                + s[4:6]
+                + "/"
+                + s[6:8]
+                + " "
+                + s[8:10]
+                + ":"
+                + s[10:12]
+                + " | "
+                + s[12:16]
+                + "/"
+                + s[16:18]
+                + "/"
+                + s[18:20]
+                + " "
+                + s[20:22]
+                + ":"
+                + s[22:24]
+            )
+
+            return super().__new__(cls, humanstr)
+
+    def serialize(self):
+        """Serialize holiday start stop values into binary values."""
+
+        data = self
+        if not re.match(
+            r"\d\d\d\d/\d\d/\d\d \d\d:\d\d \| \d\d\d\d/\d\d/\d\d \d\d:\d\d", str(data)
+        ):
+            raise ValueError(
+                "Holiday Start Stop must be in the following format:"
+                " 2021/01/01 01:01 | 2021/01/01 01:01"
+            )
+        b = (
+            data[0:4]
+            + data[5:7]
+            + data[8:10]
+            + data[11:13]
+            + data[14:16]
+            + data[19:23]
+            + data[24:26]
+            + data[27:29]
+            + data[30:32]
+            + data[33:35]
+        )
+
+        ret = b[::-1].encode()
+        return ret
+
+
+class ZONNSMARTScheduleSet(t.FixedList, item_type=t.uint8_t, length=31):
+    """Set the schedule for one or multiple days."""
+
+    def __init__(self, value, *args, **kwargs):
+        """Parse the schedule string and serialize it."""
+
+        super().__init__(*args, **kwargs)
+        if value is None:
+            return
+
+        if not isinstance(value, str):
+            raise ValueError("Only string value allowed")  # noqa: TRY004
```

Comment:
So, all errors aren't covered. Would it be possible to add tests for those? It should just be this per possible error:
```python
    with pytest.raises(
        ValueError,
        match="Only string value allowed",
    ):
        # call code here to produce the error
```

Reply:
I added tests for all those but I encountered a bug in zigpy while doing so.

Please review https://github.com/zigpy/zigpy/pull/1353 to make the tests work.

Thanks